pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/stateful/DataCollectorService.java

```diff
@@ -41,21 +42,31 @@ protected void run() throws Exception {
         log.info("Running");
         for (;;) {
             try {
-                // TODO: Get state from persistent database
-                S state = driver.initialState();
-                // Poll sensor for events.
-                final PollResponse<S> response = driver.pollEvents(state);
-                // TODO: Add samples, state atomically to persistent queue.
-                response.events.stream().forEach(event -> {
-                    try {
-                    log.trace("Adding event {}", event);
-                    persistentQueue.add(event);
-                    } catch (Exception e) {
-                        throw new RuntimeException(e);
-                    }
-                });
-                // TODO: Write state to database
-                // TODO: Commit SQL transaction
+                //Get state from persistent database
+                // S state = driver.initialState();
+                ReadingState readingState = new ReadingState(persistentQueue.connection);
+                S state = (S)readingState.getState();
+
+                try(final AutoRollback autoRollback = new AutoRollback(persistentQueue.connection))
+                {
+                    // Poll sensor for events.
+                    final PollResponse<S> response = driver.pollEvents(state);
+                    //Add samples, state atomically to persistent queue.
+                    response.events.stream().forEach(event -> {
+                        try {
+                            log.trace("Adding event {}", event);
+                            persistentQueue.addWithoutCommit(event);                        
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    });                    
+                    //Write state to database                    
+                    log.info("Last state = {}", state);
+                    readingState.updateState((String)response.state);
+                    log.info("New State = {}",readingState.getState());
```

Comment:
Do not log readingState.getState() since this will perform file I/O and slow things down unnecessarily. Instead, log response.state.

Reply:
Changed to log response.state.

Reply:
Done